packages/editor/src/extensions/code-block/highlighter.ts

```diff
@@ -124,24 +124,22 @@ export function HighlighterPlugin({
           if (!pluginState) return;
 
           const changedBlocks: Set<string> = new Set();
-          for (const blockKey in pluginState.languages) {
-            if (HIGHLIGHTED_BLOCKS.has(blockKey)) continue;
 
+          for (const blockKey in pluginState.languages) {
             const language = pluginState.languages[blockKey];
+            if (
+              HIGHLIGHTED_BLOCKS.has(blockKey) &&
+              refractor.registered(language)
+            ) {
+              continue;
+            }
+
             const languageDefinition = Languages.find(
               (l) =>
                 l.filename === language || l.alias?.some((a) => a === language)
             );
             if (!languageDefinition) continue;
 
-            if (refractor.registered(language)) {
-              if (!HIGHLIGHTED_BLOCKS.has(blockKey)) {
-                HIGHLIGHTED_BLOCKS.add(blockKey);
-                changedBlocks.add(blockKey);
-              }
-              continue;
-            }
-
```

Comment:
Why is this removed?

Reply:
This is unnecessary now as we are already checking if the language is registered in the top if statement.

Reply:
What about everything inside that if block?

Reply:
I will add it back my bad It was a mistake.